Dockerfile

```diff
@@ -30,6 +29,13 @@ RUN npm install
 # Everything for now
 COPY . .
 
+# Conditionally run npm run build based on ARG_GRAPHQL_ENDPOINT
+RUN if [ "$ARG_BUILD_ENVIRONMENT" = "development" ]; then \
```

Comment:
It's a bit confusing, but it's not really hardcoded but rather a default is provided. Look at the step `build-and-push` in `build-release-docker-hub.yml`:

```yaml
      - name: Build and push
        id: docker_build
        uses: docker/build-push-action@v2
        with:
          context: .
          file: ./Dockerfile
          platforms: linux/amd64
          push: ${{ github.event_name != 'pull_request' }}
          tags: ${{ steps.prep.outputs.tags }}
          build-args: |
            ARG_BUILD_VERSION=${{ steps.prep.outputs.version }}
            ARG_BUILD_DATE=${{ steps.prep.outputs.created }}
            ARG_BUILD_REVISION=${{ github.sha }}
            **ARG_BUILD_ENVIRONMENT=production**
          labels: |
            org.opencontainers.image.title=${{ github.event.repository.name }}
            org.opencontainers.image.description=${{ github.event.repository.description }}
            org.opencontainers.image.url=${{ github.event.repository.html_url }}
            org.opencontainers.image.source=${{ github.event.repository.clone_url }}
            org.opencontainers.image.version=${{ steps.prep.outputs.version }}
            org.opencontainers.image.created=${{ steps.prep.outputs.created }}
            org.opencontainers.image.revision=${{ github.sha }}
            org.opencontainers.image.licenses=${{ github.event.repository.license.spdx_id }}
```

Reply:
It defaults to `development` because most of our environments use that configuration (so I default to it). I override it to `production` in the `production` build.

That would mean that `Acc` is exact replice of production and we won't be able to debug there, but I guess that's not the end of the world.